src/main/kotlin/com/github/arnaudj/linkify/spi/jira/restclient/CookieStore.kt

```diff
@@ -0,0 +1,24 @@
+package com.github.arnaudj.linkify.spi.jira.restclient
+
+import okhttp3.Cookie
+import okhttp3.CookieJar
+import okhttp3.HttpUrl
+
+open class CookieStore : CookieJar {
+
+    private final val cookieStore : Set<Cookie> = hashSetOf<Cookie>();
+
+    override fun loadForRequest(url: HttpUrl): MutableList<Cookie> {
+        val  validCookies : ArrayList<Cookie> = arrayListOf<Cookie>();
+        cookieStore.forEach {
+            if(it.expiresAt() > System.currentTimeMillis()) {
```

Comment:
I'm wondering if expiry handling is complete

Does the jira endpoint return a cookie with Expires/Max-Age directive? (in "Set-Cookie: " header)

If not, afaik, this is a session cookie valid for the session time (which I expect to be capped at 8 hours)
In this case, seems expiresAt() will return "December 31, 9999", so the cookie will always be forwarded. 
Then I expect jira to reply with a 401 that we should handle by replaying the request with Authorization header, this time.

What do you think?

Refs:
https://developer.mozilla.org/fr/docs/Web/HTTP/Headers/Set-Cookie
https://developer.atlassian.com/cloud/jira/platform/jira-rest-api-cookie-based-authentication/

Reply:
i'll see that